hive-metastore/src/main/java/org/apache/iceberg/hive/HMSTablePropertyHelper.java

```diff
@@ -0,0 +1,227 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.hive;
+
+import static org.apache.iceberg.TableProperties.GC_ENABLED;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import org.apache.hadoop.hive.common.StatsSetupConst;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.hadoop.hive.metastore.api.hive_metastoreConstants;
+import org.apache.iceberg.BaseMetastoreTableOperations;
+import org.apache.iceberg.PartitionSpecParser;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.SchemaParser;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.SnapshotSummary;
+import org.apache.iceberg.SortOrderParser;
+import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.TableProperties;
+import org.apache.iceberg.relocated.com.google.common.annotations.VisibleForTesting;
+import org.apache.iceberg.relocated.com.google.common.collect.BiMap;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableBiMap;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.util.JsonUtil;
+import org.apache.parquet.hadoop.ParquetOutputFormat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HMSTablePropertyHelper {
+  private static final Logger LOG = LoggerFactory.getLogger(HMSTablePropertyHelper.class);
+
+  public static final String TABLE_TYPE_PROP = "table_type";
+  public static final String ICEBERG_TABLE_TYPE_VALUE = "iceberg";
+
+  private static final BiMap<String, String> ICEBERG_TO_HMS_TRANSLATION =
+      ImmutableBiMap.of(
+          // gc.enabled in Iceberg and external.table.purge in Hive are meant to do the same things
+          // but with different names
+          GC_ENABLED,
+          "external.table.purge",
+          TableProperties.PARQUET_COMPRESSION,
+          ParquetOutputFormat.COMPRESSION,
+          TableProperties.PARQUET_ROW_GROUP_SIZE_BYTES,
+          ParquetOutputFormat.BLOCK_SIZE);
+
+  private HMSTablePropertyHelper() {}
+
+  public static void setHmsTableParameters(
+      String newMetadataLocation,
+      Table tbl,
+      TableMetadata metadata,
+      Set<String> obsoleteProps,
+      boolean hiveEngineEnabled,
+      long maxHiveTablePropertySize,
+      String currentLocation) {
+    Map<String, String> parameters =
+        Optional.ofNullable(tbl.getParameters()).orElseGet(Maps::newHashMap);
+    Map<String, String> summary =
```

Comment:
nit: add line break before and after

Reply:
Iceberg formatter likes this way, so not modifying it. It is really sensitive on every formatting.